Develop

erised-ice opened the pull request.

erised-ice committed:
theme changing and counter done

erised-ice committed:
Typing of digits in display and clear - done

erised-ice committed:
Made concatenation of operands

erised-ice committed:
Add different operations

webestet commented on js/script.js:
```diff
@@ -1 +1,79 @@
-let apple = 1;
+let body = document.querySelector('.js-body');
```
На `body` можно не навешивать класс, потому что это всегда один элемент на странице, к которому можно обратиться через `document.body`

a-panteleeva replied:
ок

webestet commented on js/script.js:
```diff
@@ -1 +1,79 @@
-let apple = 1;
+let body = document.querySelector('.js-body');
+let themeButton = document.querySelector('.js-theme-button');
```
переменные, которые не изменяются лучше определять, использую `const`. Это сразу же создает очевидность для программиста, будет ли переменная переопределена (или же будет что-то в нее записано) ниже по коду. Если используется `const` то сразу ясно, что это непереопределяемая переменная. Большинство линтеров сейчас настроено, что `const` должен быть в приоритете при определении переменных. 

Нужно бы переименовать все переменные, которые не будут изменены в коде на `const`

a-panteleeva replied:
Поняла

webestet commented on js/script.js:
```diff
@@ -1 +1,79 @@
+let themeButton = document.querySelector('.js-theme-button');
+let countButton = document.querySelector('.js-count-button');
+let pressNumber = document.querySelector('.count-circle');
+let counter = 0;
```
// придирка

По типу данных лучше отбивать блоками переменные. Например, если ты объявляешь сначала все `Dom Elements` их в одном блоке, потом отбить пустой строкой и уже объявить переменные `let` которые служат изменяемыми данными

a-panteleeva replied:
Ок

webestet commented on js/script.js:
```diff
@@ -1 +1,79 @@
+let resultButton = document.querySelector('.js-result-button');
+
+/* Меняем тему */
+themeButton.onclick = function () {
```
Сейчас не рекомендуют использовать такие обработчики событий. тут подробнее:
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener

Лучше все события переписать на `addEventlistener` где `listener` также будет отдельной функцией, чтобы иметь возможность использовать `removeEventListener` Пример

```
const button = document.querySelector("button");
const handleButtonClick = (evt) => console.log(evt.target.textContent);

button.addEventListener(button, handleButtonClick);
```

Это самый правильный путь для работы с событиями. Можно еще также исползовать вот такой,

```
const button = document.querySelector("button");

button.addEventListener(button, (evt) => console.log(evt.target.textContent));
```
Но так не рекомендуется делать, потому что при таком подходе нельзя будет использовать `removeEventListener` так как функция с консолью не сохранена в памяти нигде. А когда ты ее выносишь отдельно, то сохраняешь ее в памяти, и потом ее можно оттуда удалить

a-panteleeva replied:
Я, пожалуй, повторю функции

a-panteleeva replied:
Я что-то вывела в отдельные функции, что-то не смогла.

webestet replied:
нужно сесть вместе и разобраться, чтобы все было вынесено в отдельные хандлеры

a-panteleeva replied:
Разобрались

webestet commented on js/script.js:
```diff
@@ -1 +1,79 @@
+let secondOperand = '';
+/**/
+
+for (let operationButton of operationButtons) {
```
`operationButtons` это массив. Коллекция `DOM Nodes`. В `JS` массивы являются мощнейшим инструментом для хранения данных с нужными методами. Когда мы работаем с массивами, мы предпочитаем работать с определенными методами
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array

Например, здесь можно использовать `forEach` вместо цикла `for`.  Не сказать, что есть особые отличия, но цикл `for` используется только тогда, когда тебе его нужно прервать. Во всех остальных случаях используются методы массива. 
```
operationsButton.forEach((operationButton) => {
 // ... code here
}) 
```

a-panteleeva replied:
Ok

webestet replied:
Решила не использовать массив с методами?

a-panteleeva replied:
Пропустила один цикл, исправила

webestet commented on js/script.js:
```diff
@@ -1 +1,79 @@
+    display.value = firstOperand / secondOperand;
+  } else if (operation === 'multiply') {
+    display.value = firstOperand * secondOperand;
+  }
```
Для подобных конструкций лучше всего использовать `switch/case` потому что он более наглядный и намного удобнее читать. `else if` практически нигде не используется. Вторым вариантом будет литерал объекта, но пока достаточно будет переписать на `switch/case`

a-panteleeva replied:
Хорошо

webestet replied:
это нужно вынести в отдельную функцию, которая будет просто возвращать результат. Туда передавать `first, second и operation`

erised-ice committed:
body class's been deleted;
the way of announcement of constants has been changed;
onclicks were changed to event.listeners;

erised-ice committed:
Switch has been implemented;

a-panteleeva reviewed (commented):
Я сделала правки по пр

webestet commented on js/script.js:
```diff
@@ -1 +1,100 @@
+})
+
+digitButtons.forEach((digitButton) => {
+  digitButton.addEventListener("click", function (event) {
```
все хэндлер функции должны быть в отдельными функциями
```
digitButton.addEventLister("click", hadleClickDigitButton)

const hadleClickDigitButton = (event) => {
  event.preventDefault();
  ....
}
```

a-panteleeva replied:
Готово
